openshift/cluster-ingress-operator
# Bug 1942657:  Ingress operator stays degraded after privateZone fixed in DNS

This bugfix addresses the BZ https://bugzilla.redhat.com/show_bug.cgi?id=1942657

Verification Steps

1. Create cluster via cluster-bot version 4.8.0

2. Start local dev - ```make run-local```

3. Verify ingress is working
```
$oc get clusteroperators/ingress

NAME      VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE
ingress   4.8.0     True        False         False      20m
```

4. Change tags.name (prefix with lmz)
```
$oc edit dns.config.openshift.io cluster

apiVersion: config.openshift.io/v1
  7 kind: DNS
  8 metadata:
  9   creationTimestamp: "2021-08-05T10:49:47Z"
 10   generation: 1
 11   name: cluster
 12   resourceVersion: "653"
 13   uid: d47dab3f-1955-418b-942b-ca002e7a3f99
 14 spec:
 15   baseDomain: ci-ln-tdn0hi2-d5d6b.origin-ci-int-aws.dev.rhcloud.com
 16   privateZone:
 17     tags:
 18       Name: lmz-ci-ln-tdn0hi2-d5d6b-kth95-int
 19       kubernetes.io/cluster/ci-ln-tdn0hi2-d5d6b-kth95: owned
 20   publicZone:
 21     id: Z2GYOLTZHS5VK
 22 status: {}

dns.config.openshift.io/cluster edited
```

5. Check ingress status has degraded
```
$ oc get clusteroperators/ingress
NAME      VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE
ingress   4.8.0     False       True          True       4m12s
```

6. Check logs 

```2021-08-05T12:32:34.414+0100	ERROR	operator.dns_controller	dns/controller.go:190	failed to publish DNS record to zone	{"record": {"dnsName":"*.apps.ci-ln-tdn0hi2-d5d6b.origin-ci-int-aws.dev.rhcloud.com.","targets":["ac3079d0e08d7428b8aa6536d7c1057e-1686588591.us-west-2.elb.amazonaws.com"],"recordType":"CNAME","recordTTL":30}, "dnszone": {"tags":{"Name":"lmz-ci-ln-tdn0hi2-d5d6b-kth95-int","kubernetes.io/cluster/ci-ln-tdn0hi2-d5d6b-kth95":"owned"}}, "error": "failed to find hosted zone for record: no matching hosted zone found"}
2021-08-05T12:32:34.414+0100	INFO	operator.dns_controller	dns/controller.go:190	skipping zone to which the DNS record is already published	{"record": {"dnsName":"*.apps.ci-ln-tdn0hi2-d5d6b.origin-ci-int-aws.dev.rhcloud.com.","targets":["ac3079d0e08d7428b8aa6536d7c1057e-1686588591.us-west-2.elb.amazonaws.com"],"recordType":"CNAME","recordTTL":30}, "dnszone": {"id":"Z2GYOLTZHS5VK"}}
2021-08-05T12:33:04.415+0100	INFO	operator.dns_controller	controller/controller.go:298	reconciling	{"request": "openshift-ingress-operator/default-wildcard"}
2021-08-05T12:33:05.384+0100	ERROR	operator.dns_controller	dns/controller.go:190	failed to publish DNS record to zone	{"record": {"dnsName":"*.apps.ci-ln-tdn0hi2-d5d6b.origin-ci-int-aws.dev.rhcloud.com.","targets":["ac3079d0e08d7428b8aa6536d7c1057e-1686588591.us-west-2.elb.amazonaws.com"],"recordType":"CNAME","recordTTL":30}, "dnszone": {"tags":{"Name":"lmz-ci-ln-tdn0hi2-d5d6b-kth95-int","kubernetes.io/cluster/ci-ln-tdn0hi2-d5d6b-kth95":"owned"}}, "error": "failed to find hosted zone for record: no matching hosted zone found"}
```

7. Change the dns private zone config to original
```
$oc edit dns.config.openshift.io cluster
   
apiVersion: config.openshift.io/v1
kind: DNS
metadata:
  creationTimestamp: "2021-08-05T10:49:47Z"
  generation: 1
  name: cluster
  resourceVersion: "653"
  uid: d47dab3f-1955-418b-942b-ca002e7a3f99
spec:
  baseDomain: ci-ln-tdn0hi2-d5d6b.origin-ci-int-aws.dev.rhcloud.com
  privateZone:
    tags:
      Name: ci-ln-tdn0hi2-d5d6b-kth95-int
      kubernetes.io/cluster/ci-ln-tdn0hi2-d5d6b-kth95: owned
  publicZone:
    id: Z2GYOLTZHS5VK
status: {}

dns.config.openshift.io/cluster edited
```

8. Verify logs
```
2021-08-05T12:35:09.903+0100	INFO	operator.dns_controller	dns/controller.go:190	skipping zone to which the DNS record is already published	{"record": {"dnsName":"*.apps.ci-ln-tdn0hi2-d5d6b.origin-ci-int-aws.dev.rhcloud.com.","targets":["ac3079d0e08d7428b8aa6536d7c1057e-1686588591.us-west-2.elb.amazonaws.com"],"recordType":"CNAME","recordTTL":30}, "dnszone": {"tags":{"Name":"ci-ln-tdn0hi2-d5d6b-kth95-int","kubernetes.io/cluster/ci-ln-tdn0hi2-d5d6b-kth95":"owned"}}}
2021-08-05T12:35:09.903+0100	INFO	operator.dns_controller	dns/controller.go:190	skipping zone to which the DNS record is already published	{"record": {"dnsName":"*.apps.ci-ln-tdn0hi2-d5d6b.origin-ci-int-aws.dev.rhcloud.com.","targets":["ac3079d0e08d7428b8aa6536d7c1057e-1686588591.us-west-2.elb.amazonaws.com"],"recordType":"CNAME","recordTTL":30}, "dnszone": {"id":"Z2GYOLTZHS5VK"}}
```

9. Check ingress status
```
$ oc get clusteroperators/ingress
NAME      VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE
ingress   4.8.0     True        False         False      83s
```
